test/integration/puppetlabs/general_puppet/general_puppet_int_test.clj

```diff
@@ -38,3 +37,27 @@
      (testing "calling generate successfully executes shell command"
        (let [catalog (get-catalog)]
          (is (catalog-contains? catalog "Notify" "this command echoes a thing\n")))))))
+
+(deftest ^:integration code-id-request-test
+  (testing "code id is added to the request body for catalog requests"
+    ; As we have set code-id-command to echo, the code id will
+    ; be the result of running `echo $environment`, which will
+    ; be production here.
+    (bootstrap/with-puppetserver-running
+     app {:jruby-puppet
+          {:max-active-instances num-jrubies}
+          :versioned-code
+          {:code-id-command (script-path "echo")}}
```

Comment:
I think I'll be adding a commit to rearrange some common shared test bits, so this will get cleaned up in that refactor.